rules-tests/DeadCode/Rector/StaticCall/RemoveParentCallWithoutParentRector/Fixture/OtherNamespace/parent_same_namespace_aliased_parent_parent_class.php.inc

```diff
@@ -0,0 +1,9 @@
+<?php
+
+namespace Rector\Tests\DeadCode\Rector\StaticCall\RemoveParentCallWithoutParentRector\Fixture\OtherNamespace;
+
+class ParentSameNamespaceAliasedParentParentClass {
```

Comment:
You can place parent class in `Source` directory, along update namespace.

Actually, it should already covered, you was probably has issue with autoloading, see the 

https://github.com/rectorphp/rector/blob/main/docs/static_reflection_and_autoload.md#troubleshooting

Reply:
Oh, same namespace, it probably correct in current Fixture directory:+1:

Reply:
> You can place parent class in Source directory, along update namespace.

I think this would be the best choice, as the class is only for reflection, not for change.

@bradjones1 Could you move the parent class to `/Source` directory and only re-use it in the fixture?

Reply:
@TomasVotruba the issue is it has same namespace + aliased, so possibly just combine in single file.

Reply:
it possibly parent check not autoload need to be skipped needs to apply on aliased name

https://github.com/rectorphp/rector-src/blob/236b0ffe16821d82446dffd3603e5ccb8cba8621/rules/DeadCode/Rector/StaticCall/RemoveParentCallWithoutParentRector.php#L131-L133

so when autoload is overlapped (child loaded first, then parent), it just skipped.

Reply:
Ok, I tried locally, and it seems working ok, even run test to the whole `rules-tests` directory:

```bash
➜  vendor/bin/paratest rules-tests 
ParaTest v6.6.1 upon PHPUnit 9.5.21 #StandWithUkraine

.............................................................   61 / 3368 (  1%)
.............................................................  122 / 3368 (  3%)
.............................................................  183 / 3368 (  5%)
.............................................................  244 / 3368 (  7%)
.............................................................  305 / 3368 (  9%)
.............................................................  366 / 3368 ( 10%)
.............................................................  427 / 3368 ( 12%)
.............................................................  488 / 3368 ( 14%)
.............................................................  549 / 3368 ( 16%)
.............................................................  610 / 3368 ( 18%)
.............................................................  671 / 3368 ( 19%)
.............................................................  732 / 3368 ( 21%)
.............................................................  793 / 3368 ( 23%)
.............................................................  854 / 3368 ( 25%)
.............................................................  915 / 3368 ( 27%)
.............................................................  976 / 3368 ( 28%)
............................................................. 1037 / 3368 ( 30%)
............................................................. 1098 / 3368 ( 32%)
............................................................. 1159 / 3368 ( 34%)
............................................................. 1220 / 3368 ( 36%)
............................................................. 1281 / 3368 ( 38%)
............................................................. 1342 / 3368 ( 39%)
............................................................. 1403 / 3368 ( 41%)
............................................................. 1464 / 3368 ( 43%)
............................................................. 1525 / 3368 ( 45%)
............................................................. 1586 / 3368 ( 47%)
............................................................. 1647 / 3368 ( 48%)
............................................................. 1708 / 3368 ( 50%)
............................................................. 1769 / 3368 ( 52%)
............................................................. 1830 / 3368 ( 54%)
............................................................. 1891 / 3368 ( 56%)
............................................................. 1952 / 3368 ( 57%)
............................................................. 2013 / 3368 ( 59%)
............................................................. 2074 / 3368 ( 61%)
............................................................. 2135 / 3368 ( 63%)
............................................................. 2196 / 3368 ( 65%)
............................................................. 2257 / 3368 ( 67%)
............................................................. 2318 / 3368 ( 68%)
............................................................. 2379 / 3368 ( 70%)
............................................................. 2440 / 3368 ( 72%)
............................................................. 2501 / 3368 ( 74%)
............................................................. 2562 / 3368 ( 76%)
............................................................. 2623 / 3368 ( 77%)
............................................................. 2684 / 3368 ( 79%)
............................................................. 2745 / 3368 ( 81%)
............................................................. 2806 / 3368 ( 83%)
............................................................. 2867 / 3368 ( 85%)
............................................................. 2928 / 3368 ( 86%)
............................................................. 2989 / 3368 ( 88%)
............................................................. 3050 / 3368 ( 90%)
............................................................. 3111 / 3368 ( 92%)
............................................................. 3172 / 3368 ( 94%)
............................................................. 3233 / 3368 ( 95%)
............................................................. 3294 / 3368 ( 97%)
............................................................. 3355 / 3368 ( 99%)
.............                                                 3368 / 3368 (100%)

Time: 00:49.843, Memory: 34.00 MB

```

So, its definatelly an autoload overlapped issue, you can try autoload troubleshooting that documented at docs:

https://github.com/rectorphp/rector/blob/main/docs/static_reflection_and_autoload.md#troubleshooting

for this kind of issue.

Reply:
That would explain why it's so frustratingly intermittent - thank you and I'm sorry I didn't catch the autoloading aspect earlier.

Reply:
I think it is close-able then, as it only be able to be resolved via the autoloading hack as mentioned in the doc.

Reply:
Agreed, though I'm confused why it would be so intermittent.